site/docs/tools/iceberg/spark.md

```diff
@@ -292,22 +292,22 @@ To read a Nessie table in iceberg simply:
 
 The examples above all use the default branch defined on initialisation. There are several ways to reference specific
 branches or hashes from within a read statement. We will take a look at a few now from pyspark3, the rules are the same
-across all environments though. The general pattern is `<table>@<branch>`. Table must be present and either
+across all environments though. The general pattern is `<table>@<branch>` or `<table>@<branch>#<hash>`. Table must be present and either
 branch and/or hash are optional. We will throw an error if branch or hash don't exist.
 Branch or hash references in the table name will override passed `option`s and the settings in the
 Spark/Hadoop configs.
 
 ``` python linenums="1"
 # read from branch dev
 spark.read().format("iceberg").load("testing.region@dev")
-# read specifically from hash
-spark.read().format("iceberg").load("testing.region@<hash>")
+# read specifically from hash in dev branch
+spark.read().format("iceberg").load("testing.region@dev#<hash>")
```

Comment:
@dimas-b I think this should be fixed in site/docs/tools/iceberg/flink.md page as well? WDYT?

Reply:
We may want to fix code and make the old syntax working again per #7556... not sure about this change ATM.

Reply:
Sounds good to me, I would like to take it up.